File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.js
function solution(phone_number) {
    var answer = '';
    if(phone_number.length === 4){
        return phone_number
    }
    let changed_numbers = phone_number.slice(0, -4).split("").map(e => e = "*").join("")
    let last_numbers = phone_number.slice(-4)
    
    answer += changed_numbers
    answer += last_numbers
    
    return answer;
}

File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.java
import java.util.*;

class Solution {
    public String solution(String phone_number) {

        StringBuilder sb = new StringBuilder();
        
        String[] phone = phone_number.split("");
        for(int i = 0; i < phone.length - 4; i++){
           phone[i] = "*";
            sb.append(phone[i]);
        }
        
        for(int i = phone.length - 4; i < phone.length; i++){
            sb.append(phone[i]);
        }

        return sb.toString();
    }
}